### PR TITLE
refactor: Refactor how manifest flags are set

### PR DESCRIPTION
**Pull request by Gudahtt**
## **Description**

The pieces of `set-manifest-flags.ts` related to _getting_ flags have been moved to the new `get-manifest-flags.ts` module. This module will be used in a later PR by a script run during a CI workflow.

The migrated steps were also made asynchronous so that the asynchronous steps could be run in parallel rather than blocking the process.

Documentation has been added to the `ManifestFlags` type as well, in preparation for adding a new property in the next PR.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28686?quickstart=1)

## **Related issues**

Related to https://github.com/MetaMask/metamask-extension/issues/28685

## **Manual testing steps**

N/A, no functional changes.

## **Screenshots/Recordings**

N/A

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by Gudahtt** [comment]
This refactor is in support of https://github.com/MetaMask/metamask-extension/pull/28687

**Review by HowardBraham — COMMENTED** [review]
Mostly just rewriting the comments to be clearer, and one actual bug.

Also, `check-optional-e2e-trigger.ts` should be run through Prettier.

File: development/lib/get-manifest-flag.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+ * To use this feature, add a line to your commit message like:
+ * `flags = {"sentry": {"tracesSampleRate": 0.1}}`
+ * (must be valid JSON)
+ *
+ * @returns Any manifest flags found in the commit message
+ */
+async function getFlagsFromGitMessage(): Promise<ManifestFlags> {
+  const gitMessage = (
+    await exec(`git show --format='%B' --no-patch "HEAD"`)
+  ).toString();
```
[the remark this answers is not in the document]
Comment (HowardBraham) [reply development/lib/get-manifest-flag.ts:86]: No, it was not broken before.  It was previously `execSync().toString()`

[execSync](https://nodejs.org/api/child_process.html#child_processexecsynccommand-options) `Returns: <Buffer> | <string> The stdout from the command.`

[exec](https://nodejs.org/api/child_process.html#child_processexeccommand-options-callback) callback returns
```
error <Error>
stdout <string> | <Buffer>
stderr <string> | <Buffer>
```

[the remark this answers is not in the document]
Comment (Gudahtt) [reply development/lib/get-manifest-flag.ts:86]: 🤦 I should have read the docs more carefully. Great catch, thanks!

**Review by HowardBraham — DISMISSED** [review]

File: development/lib/get-manifest-flag.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  return regexSearchForFlags(gitMessage, 'git message');
+}
+
+/**
+ * Get any manifest flags found in the PR body and git message.
+ *
+ * @returns Any manifest flags found
+ */
+export async function getManifestFlags(): Promise<ManifestFlags> {
```
Comment (davidmurdoch) [inline development/lib/get-manifest-flag.ts:95]: We now have two functions in the code base named `getManifestFlags` that both end up getting the same flags, but in different ways (runtime vs compile time) :-/ One is async (this one), and the other must stay synchronous as it is designed to be blocking and used on app initialization.

Think you could rename one or the other?

Comment (Gudahtt) [reply development/lib/get-manifest-flag.ts:95]: Good point. Seems better to rename this one. Maybe `fetchManifestFlags` or `createManifestFlags`?

Comment (davidmurdoch) [reply development/lib/get-manifest-flag.ts:95]: Either is fine with me!

Comment (HowardBraham) [reply development/lib/get-manifest-flag.ts:95]: @Gudahtt @davidmurdoch @DDDDDanica I think this one should be called `fetchManifestFlagsFromGitAndPR`

**Review by HowardBraham — DISMISSED** [review]

**Review by HowardBraham — DISMISSED** [review]

**Review by HowardBraham — APPROVED** [review]

**Review by DDDDDanica — APPROVED** [review]

**Comment by DDDDDanica** [comment]
LGTM !

**Review by davidmurdoch — APPROVED** [review]

**Review by Gudahtt — COMMENTED** [review]
Thanks Howard for renaming, LGTM

**Pull request merged**